.github/workflows/workflow.yaml

```diff
@@ -53,3 +53,33 @@ jobs:
   #        uses: codecov/codecov-action@v3
   #        with:
   #          token: ${{ secrets.CODECOV_TOKEN }}
+  publish:
+    name: Publish to NPM
+    if: github.ref == 'refs/heads/main'
+    needs:
+      - check-for-sh
+      - lint
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      # actions/checkout MUST come before auth
+      - uses: actions/checkout@v3
+      - id: auth
+        uses: google-github-actions/auth@v0
+        with:
+          credentials_json: ${{ secrets.MAINNET_SERVICE_ACCOUNT_KEY }}
+      - id: google-secrets
+        uses: google-github-actions/get-secretmanager-secrets@v0
+        with:
+          secrets: |-
+            NPM_TOKEN:projects/1027349420744/secrets/NPM_TOKEN
```

Comment:
this is cool. does it need a version?

Reply:
The version is optional, I now realize there's an even [shorter form](https://github.com/google-github-actions/get-secretmanager-secrets#inputs):

```
# Long form
projects/<project-id>/secrets/<secret-id>/versions/<version-id>

# Long form - "latest" version
projects/<project-id>/secrets/<secret-id>

# Short form
<project-id>/<secret-id>/<version-id>

# Short form - "latest" version
<project-id>/<secret-id>
```